src/lib/enrollment-submission-handler.js

```diff
@@ -0,0 +1,76 @@
+import { getErrorMessages } from '.';
+import {
+  createOrUpdateEnrollment,
+  deleteEnrollment,
+  computeNextEnrollmentState,
+} from '../services/enrollments';
+
+export const handleEnrollmentSubmission = async (
+  actionConfiguration,
+  setIntendedAction,
+  enrollment,
+  waitForUserInteractionInPrompt,
+  updateEnrollment
+) => {
+  const resultMessages = {
+    errorMessages: [],
+    successMessages: [],
+    redirectToHome: false,
+  };
+
+  const action = actionConfiguration.id;
+
+  try {
+    let comment = null;
+    let commentFullEditMode = null;
+
+    if (actionConfiguration.promptForComment) {
+      try {
+        setIntendedAction(action);
+
+        const actionMessage = await waitForUserInteractionInPrompt;
+        comment = actionMessage.message;
+        commentFullEditMode = actionMessage.fullEditMode;
+      } catch (e) {
+        return resultMessages;
+      }
+    }
+
+    let enrollmentId = enrollment.id;
+
+    if (enrollment.acl.update) {
+      const newEnrollment = await createOrUpdateEnrollment({
+        enrollment,
+      });
+      updateEnrollment(newEnrollment);
+      enrollmentId = newEnrollment.id;
+
+      resultMessages.successMessages.push('Votre demande a été sauvegardée.');
+    }
+
+    if (action === 'update') {
+      return resultMessages;
+    }
+
+    if (action === 'destroy') {
+      await deleteEnrollment({ id: enrollmentId });
+    }
+
+    if (actionConfiguration.needsToComputeNextEnrollmentState) {
```

Comment:
I like the fact that this is now a parameters, but I can't manage to understand the meaning of this. Can we talk about this ?

Reply:
Maybe my confusion is partially due to NextEnrollment being a name used elsewhere in the codebase.